log-collector-script/linux/eks-log-collector.sh

```diff
@@ -483,20 +489,22 @@ get_k8s_info() {
   fi
 
   case "${INIT_TYPE}" in
-    systemd)
-      timeout 75 journalctl --unit=kubelet --since "${DAYS_10}" > "${COLLECT_DIR}"/kubelet/kubelet.log
+    systemd | snap)
+      timeout 75 snap list kubelet-eks > /dev/null 2>&1
+      if [ 0 -eq $? ]; then # Check if previous command was successful.
+        timeout 75 snap logs kubelet-eks -n all > "${COLLECT_DIR}"/kubelet/kubelet.log
 
-      systemctl cat kubelet > "${COLLECT_DIR}"/kubelet/kubelet_service.txt 2>&1
+        timeout 75 snap get kubelet-eks > "${COLLECT_DIR}"/kubelet/kubelet-eks_service.txt 2>&1
+      else
+        timeout 75 journalctl --unit=kubelet --since "${DAYS_10}" > "${COLLECT_DIR}"/kubelet/kubelet.log
 
-      cp --force --recursive --dereference /etc/kubernetes/kubelet/config.json "${COLLECT_DIR}"/kubelet/config.json 2> /dev/null
-      cp --force --recursive --dereference /etc/kubernetes/kubelet/config.json.d "${COLLECT_DIR}"/kubelet/config.json.d 2> /dev/null
+        systemctl cat kubelet > "${COLLECT_DIR}"/kubelet/kubelet_service.txt 2>&1
 
-      cp --force --recursive --dereference /etc/kubernetes/kubelet/kubelet-config.json "${COLLECT_DIR}"/kubelet/kubelet-config.json 2> /dev/null
-      ;;
-    snap)
-      timeout 75 snap logs kubelet-eks -n all > "${COLLECT_DIR}"/kubelet/kubelet.log
+        cp --force --recursive --dereference /etc/kubernetes/kubelet/config.json "${COLLECT_DIR}"/kubelet/config.json 2> /dev/null
+        cp --force --recursive --dereference /etc/kubernetes/kubelet/config.json.d "${COLLECT_DIR}"/kubelet/config.json.d 2> /dev/null
 
-      timeout 75 snap get kubelet-eks > "${COLLECT_DIR}"/kubelet/kubelet-eks_service.txt 2>&1
+        cp --force --recursive --dereference /etc/kubernetes/kubelet/kubelet-config.json "${COLLECT_DIR}"/kubelet/kubelet-config.json 2> /dev/null
+      fi
       ;;
```

Comment:
`INIT_TYPE` is set by checking if `systemctl` or `snap` is on the `PATH`, so I don't think this block actually changes any logic

Reply:
The change is that before if `INIT_TYPE` is `snap` it collects the kubelet logs use `snap get kubelet-eks`. In the case that you are on the Ubuntu optimized ami that is the source of the kubelet so the logs are correctly retrieved.  If you are running on a different type of Ubuntu host, with snap, but not the `kubelet-eks` snap installed you do not capture any kubelet logs.  In this sceanrio, we want to fallback to using journalctl to collect the kubelet logs as if it was just a normal `systemd` `INIT_TYPE`.